packages/dds/tree/src/tree/anchorSet.ts

```diff
@@ -517,13 +529,15 @@ class PathNode implements UpPath {
      * the caller must ensure that the reference from child to parent is also removed (or the child is no longer used).
      */
     public removeChild(child: PathNode): void {
-        assert(this.status === Status.Alive, 0x40e /* PathNode must be alive */);
+        // It's possible for this node to be `Dead` because we don't remove the children of path nodes
+        // that correspond to deleted content.
+        assert(this.status !== Status.Disposed, "PathNode must not be disposed");
```

Comment:
The documentation on Status.Dead really makes it sound like this should not be allowed on dead nodes.

Why would we want to call this to update a dead node? The contents of dead nodes are illegal to access (so we can't update them, and we also don't care what they are.

Is the issue that freeing a dead node can try and update its parent?
Freeing a dead node should just check its already dead, and if so set the state from Dead to Free (and we should clarify the documentation on State.Dead to say this is allowed).

I think its ok if some interior nodes of deleted trees never get marked as free since doing so isn't needed to avoid leaking them, its just for double free checking.

Reply:
The reason I see for maintaining Dead nodes is that they may be revived (we don't support this yet, but we will soon). When that happens, we want the revived node's children to accurately reflect the anchors being tracked. If we don't do this, then we have to tell client code never to forget an anchor to a node that's under a dead node that might be revived.
I agree this goes against the documentation for the Dead state, and I think the name "Dead" is not great here.
Perhaps we should name it "Limbo" or "Ghost".

Reply:
I think for now deciding we don't touch the internals of dead nodes at all and make no guarantees about them  (other than the status) leaves as much room as possible for up-coming field kind specific handling of dead nodes.